aws/resource_aws_cognito_user_pool_domain.go

```diff
@@ -163,32 +140,91 @@ func resourceAwsCognitoUserPoolDomainDelete(d *schema.ResourceData, meta interfa
 		return err
 	}
 
+	err = waitForUserPoolDomainDelete(conn, d.Id())
+	return err
+}
+
+func resourceAwsCognitoUserPoolDomainUpdate(d *schema.ResourceData, meta interface{}) error {
```

Comment:
There needs to be a check in here to see if the `certificate_arn` has changed

it's usually along the lines of

```
if d.HasChange("certificate_arn")
```

Reply:
Interesting -- I see this pattern used in some, but not all of the resources, e.g. the domain pool doesn't do this. Seems better to check, so I'll definitely do that.